docs/Users/Tutorials/External_Bears.rst

```diff
@@ -0,0 +1,511 @@
+            "origin": origin,
+            "message": message,
+            "affected_code": [{
+                    "file": input["filename"],
+                    "start": {
+                        "column": column,
+                        "file": input["filename"],
+                        "line": line
+                    },
+                    "end": {
+                        "column": column+6,
+                        "file": input["filename"],
+                        "line": line
+                    }
+                }],
+            "severity": severity
+        };
+        out["results"].push(result)
+    };
+
+The last part is to iterate over the lines and check for ``"coala"`` spelled
+with a capital ``"C"``. The final source should look like this:
+
+::
+
+    var input = "";
+    var out = {};
+    var origin;
+
+    console.log = (msg) => {
+        process.stdout.write(`${msg}\n`);
+    };
+
+    init_results = (bear_name) => {
+        origin = bear_name;
+        out["results"] = [];
+    };
+
+    add_result = (message, line, column, severity) => {
+        var result = {
+            "origin": origin,
+            "message": message,
+            "affected_code": [{
+                    "file": input["filename"],
+                    "start": {
+                        "column": column,
+                        "file": input["filename"],
+                        "line": line
+                    },
+                    "end": {
+                        "column": column+6,
+                        "file": input["filename"],
+                        "line": line
+                    }
+                }],
+            "severity": severity
+        };
+        out["results"].push(result)
+    };
+
+    process.stdin.setEncoding('utf8');
+
+    process.stdin.on('readable', () => {
+        var chunk = process.stdin.read();
+        if (chunk !== null) {
+            input += chunk;
+        }
+    });
+
+    process.stdin.on('end', () => {
+        input = JSON.parse(input);
+        init_results("coalaCheckBear");
+        for (i in input["file"]) {
+            var line = input["file"][i];
+            var found = line.indexOf("Coala");
+            while (found != -1) {
+                add_result("Did you mean 'coala'?", parseInt(i)+1, found+1, 2);
+                found = line.indexOf("Coala", found+1)
+            }
+        }
+        console.log(JSON.stringify(out));
+    });
+
+And we are pretty much done. In order to run this Bear we do not need to add the
+source code to the path because the binary being run is ``node``. Although there
+is a problem: the argument we supplied will be looked up only in the current
+directory. To fix this we can add the full path of the ``.js`` file in the
+argument list. In this case we just run the bear from the same directory as
+``coalaCheck.js``. The code for this example can be found
+`here <https://github.com/Redridge/coalaCheckBear-js>`__.
+
+The JSON Spec
```

Comment:
It is used only for external bears though
I put it here and use links to it (see C++ tutorial)